include/jemalloc/internal/thread_event.h

```diff
@@ -217,12 +217,56 @@ te_ctx_get(tsd_t *tsd, te_ctx_t *ctx, bool is_alloc) {
 	}
 }
 
+/*
+ * The lookahead functionality facilitates events to be able to lookahead, i.e.
+ * without touching the event counters, to determine whether an event would be
+ * triggered.  The event counters are not advanced until the end of the
+ * allocation / deallocation calls, so the lookahead can be useful if some
+ * preparation work for some event must be done early in the allocation /
+ * deallocation calls.
+ *
+ * Currently only the profiling sampling event needs the lookahead
+ * functionality, so we don't yet define general purpose lookahead functions.
+ *
+ * Surplus is a terminology referring to the amount of bytes beyond what's
+ * needed for triggering an event, which can be a useful quantity to have in
+ * general when lookahead is being called.
+ */
+
+JEMALLOC_ALWAYS_INLINE bool
+te_prof_sample_event_lookahead_surplus(tsd_t *tsd, size_t usize,
+    size_t *surplus) {
+	if (surplus != NULL) {
+		/*
+		 * This is a dead store: the surplus will be overwritten before
+		 * any read.  The initialization suppresses compiler warnings.
+		 * Meanwhile, using usize to initialize is good for debugging
+		 * purpose, because a valid surplus value is strictly less than
+		 * usize.
+		 */
+		*surplus = usize;
```

Comment:
If I might suggest: I'm not sure I would spot that `*surplus == usize` is invalid while debugging. Some other garbage constant (e.g. `0x99999999` or `0xdeadbeef` or something) OTOH, I would.

Reply:
Makes sense. Use `SIZE_MAX` instead. (Strictly speaking `0x99999999` and `0xdeadbeef` can be valid `surplus` values.)